include/r3.h

```diff
@@ -20,6 +20,7 @@
 
 #if !defined(bool) && !defined(__cplusplus)
 typedef unsigned char bool;
+#    define bool bool
```

Comment:
This define looks like it doesn't do anything. Is it just for other ifdefs later on? Perhaps add a clarifying comment.

In str_array.h, there is similar code but this define is not there. Is that intentional?

Reply:
This is to avoid re-defining the typedef bool in the case where we dont have stdbool.h.
I'll add a comment about it and make it similar in str_array.h.
I'm not really sure which scenarios we don't have stdbool.h available, the project uses c99 which has boolean support. Maybe Windows/Borland C has some other handling.